src/Hazelcast.Net/DistributedObjects/ReliableTopicEventHandler.cs

```diff
@@ -48,25 +48,47 @@ public ReliableTopicEventHandler<T> Message(Func<IHReliableTopic<T>, ReliableTop
         return this;
     }
     
-    // Disposed event is single, no chaining.
+    /// <summary>
+    /// Sets the handler which runs after the subscription disposed or canceled.
+    /// </summary>
+    /// <param name="handler">The handler.</param>
+    public ReliableTopicEventHandler<T> Terminated(Action<IHReliableTopic<T>, ReliableTopicMessageEventArgs<T>> handler)
+    {
+        if (handler is null) throw new ArgumentNullException(nameof(handler));
+        Add(new ReliableTopicTerminatedEventHandler<T>(handler));
+        return this;
+    }
+
+    /// <summary>
+    /// Sets the handler which runs after the subscription disposed or canceled.
+    /// </summary>
+    /// <param name="handler">The handler.</param>
+    public ReliableTopicEventHandler<T> Terminated(Func<IHReliableTopic<T>, ReliableTopicMessageEventArgs<T>, ValueTask> handler)
+    {
+        if (handler is null) throw new ArgumentNullException(nameof(handler));
+        Add(new ReliableTopicTerminatedEventHandler<T>(handler));
+        return this;
+    }
+    
+    // Exception event is single, no chaining.
     
     /// <summary>
-    /// Sets the handler which runs when the subscription is disposed.
+    /// Sets the handler when runs on exception.
```

Comment:
```suggestion
    /// Sets the handler which runs when an exception is thrown WHERE??
```

Reply:
note: same for the other overload. `WHERE` = what can trigger an exception? when a message event handler throws? then we need `<remarks>` to explain: `ReliableTopicExceptionEventArgs` has a `Terminate` property which is true (false?) by default, meaning that after this exception event is handled, the subscription is terminated (and the terminated event is raised). If `Terminate` is set to false then all further handlers for *this* message are skipped, but further messages will raise events.

Reply:
More explanation added at https://github.com/hazelcast/hazelcast-csharp-client/pull/825/commits/29bfc61d64d9fcf6984c77fe142aac9fa8d43173